paywall/src/components/content/CheckoutContent.tsx

```diff
@@ -84,6 +92,20 @@ export default function CheckoutContent() {
     defaultValue: false,
     getValue: (val: any) => !!val,
   })
+
+  // This listener is used only for the side effect of closing the overlay when a purchase is rejected.
+  useListenForPostMessage({
+    type: POST_MESSAGE_ERROR,
+    defaultValue: undefined,
+    getValue: (payload: any) => {
+      if (payload === 'purchase failed') {
```

Comment:
As discussed, we may not even need to check the payload and assume that, if an error happened, it must be because the user did something on their wallet?